lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -356,6 +286,77 @@ class IntegrationTestsManager {
   }
 }
 
+/// Interface for collecting arguments to give `flutter drive` to run the
+/// integration tests.
+abstract class IntegrationArguments {
+  IntegrationArguments();
+
+  factory IntegrationArguments.fromBrowser(String browser) {
+    if (browser == 'chrome') {
+      return ChromeIntegrationArguments();
+    } else if (browser == 'safari' && io.Platform.isMacOS) {
+      return SafariIntegrationArguments();
+    } else {
+      throw StateError(unsupportedConfigurationWarning);
+    }
+  }
+
+  List<String> getTestArguments(String testName, String mode);
+
+  String getStatementToRun(String testName, String mode);
+}
+
+/// Arguments to give `flutter drive` to run the integration tests on Chrome.
+class ChromeIntegrationArguments extends IntegrationArguments {
+  List<String> getTestArguments(String testName, String mode) {
+    return <String>[
+      'drive',
+      '--target=test_driver/${testName}',
+      '-d',
+      'web-server',
+      '--$mode',
+      '--browser-name=chrome',
+      if (isLuci) '--chrome-binary=${preinstalledChromeExecutable()}',
+      if (isLuci) '--headless',
+      '--local-engine=host_debug_unopt',
+    ];
+  }
+
+  String getStatementToRun(String testName, String mode) {
```

Comment:
This should be just `=> 'flutter ${getTestArguments(testName, mode).join(' ')}'`

Reply:
But it differs depending on where it runs. Check next lines. There is an if statement for LUCI. Trying to squeeze this method like that will make it less readable.

Reply:
changed the one in the other class (SafariIntegrationArguments).